Makefile

```diff
@@ -6,12 +6,21 @@ OBJ := start.o \
 
 CFLAGS := -O0 -ffreestanding -nostdlib -Wall -Wextra -g
 
+# This lets us override CC and LD with environment variables
+ifneq '$(CC)' ''
+	CC := $(CC)
+endif
+
+ifneq '$(LD)' ''
+	LD := $(LD)
+endif
+
 all: $(KERNBIN)
 
 $(KERNBIN): $(OBJ)
 	$(LD) -T $(LDSCRIPT) -o $@ $^
 
-.PHONY: clean qemu qemu-gdb
+.PHONY: clean docker-image docker-qemu docker-qemu qemu qemu-gdb
```

Comment:
I think you put an extra `docker-qemu` here.

Is it possible to add a `docker-qemu-gdb` target?

Reply:
thanks! Yeah probably, i need to think about the best way to do that but i should be able to

Reply:
oh you mean to run qemu with the right flags to later attach with gdb and not have a make target for starting gdb. I can add both of those i think

Reply:
Yeah, running `gdb` can be manual, but a target for telling `qemu` to expose a debug port and to wait for the debugger to attach is what I'm after.